ParquetClassLibrary/Characters/BeingStatus.cs

```diff
@@ -1,37 +1,130 @@
+        [JsonProperty(PropertyName = "in_knwonCraftingRecipes")]
+        public readonly List<EntityID> KnownCraftingRecipes = new List<EntityID>();
+
+        /// <summary>The <see cref="Sandbox.RoomRecipe"/>s that this <see cref="Character"/> knows.</summary>
+        [JsonProperty(PropertyName = "in_knownRoomRecipes")]
+        public readonly List<EntityID> KnownRoomRecipes = new List<EntityID>();
+
+        /// <summary>The <see cref="NPC"/>s that this <see cref="Character"/> has met.</summary>
+        [JsonProperty(PropertyName = "in_knownCharacters")]
+        public readonly List<EntityID> KnownCharacters = new List<EntityID>();
+
+        /// <summary>The <see cref="Critter"/>s that this <see cref="Character"/> has encountered.</summary>
+        [JsonProperty(PropertyName = "in_knownCritters")]
+        public readonly List<EntityID> KnownCritters = new List<EntityID>();
+
+        /// <summary>The parquets that this <see cref="Character"/> has analyzed.</summary>
+        [JsonProperty(PropertyName = "in_knownParquets")]
+        public readonly List<EntityID> KnownParquets = new List<EntityID>();
 
         #region Initialization
         /// <summary>
         /// Initializes a new instance of the <see cref="BeingStatus"/> class.
         /// </summary>
-        /// <param name="in_beingDefinition">The parquets whose status this instance is tracking.</param>
-        /// <param name="in_currentLocation">Whether or not the <see cref="Floor"/> associated with this status has been dug out.</param>
-        /// <param name="in_currentBehavior">The <see cref="Behavior"/> currently governing .</param>
+        /// <param name="in_beingDefinition">The <see cref="Being"/> whose status is being tracked.</param>
+        /// <param name="in_position">The <see cref="Location"/> the tracked <see cref="Being"/> occupies.</param>
+        /// <param name="in_spawnAt">The <see cref="Location"/> the tracked <see cref="Being"/> will next spawn at.</param>
+        /// <param name="in_currentBehavior">The <see cref="Behavior"/> currently governing the tracked <see cref="Being"/>.</param>
+        /// <param name="in_buildingSpeed">The time it takes the tracked <see cref="Being"/> to place new parquets.</param>
+        /// <param name="in_modificationSpeed">The time it takes the tracked <see cref="Being"/> to modify existing parquets.</param>
+        /// <param name="in_gatheringSpeed">The time it takes the tracked <see cref="Being"/> to gather existing parquets.</param>
+        /// <param name="in_movementSpeed">The time it takes the tracked <see cref="Being"/> to walk from one <see cref="Location"/> to another.</param>
         [JsonConstructor]
-        public BeingStatus(Being in_beingDefinition, Location in_currentLocation, Behavior in_currentBehavior)
+        public BeingStatus(Being in_beingDefinition, Location in_position, Location in_spawnAt,
```

Comment:
Could this class use null default parameters for the new Lists? There's a good chance that the Player is the only one that needs a few of these.

Reply:
Mmm, that is a very good consideration.  Fixed!